### PR TITLE
Versioned problem example

### PR DESCRIPTION
**Pull request by MissiaL**
I started to make our schema versioned, but ran into a problem

We use a versioning class that removes fields depending on the version.

And unfortunately this does not work in schema build. I gave an example of how this works with us. Could you see why this might not work? I tried to find the reason, then I could not. 
Thanks for the help

### CODE REVIEW

**Comment by MissiaL** [comment]
This problem is part of https://github.com/tfranzel/drf-spectacular/issues/54

File: tests/test_versioning.py [hunk 1]
```diff
@@ -44,24 +96,26 @@ def list(self, request, *args, **kwargs):
...
 def test_url_path_versioning(no_warnings, viewset_cls, version):
     router = routers.SimpleRouter()
     router.register('x', viewset_cls, basename='x')
     generator = SchemaGenerator(
-        patterns=[re_path(r'^(?P<version>[v1|v2]+)/', include((router.urls, 'x')))],
+        patterns=[re_path(r'^(?P<version>[v1|v2|3]+)/', include((router.urls, 'x')))],
         api_version=version,
     )
     schema = generator.get_schema(request=None, public=True)
-    assert_schema(schema, f'tests/test_versioning_{version}.yml')
```
Comment (tfranzel) [inline tests/test_versioning.py:64]: travis will probably depend on proper path `tests/`

File: tests/test_versioning.py [hunk 2]
```diff
@@ -22,11 +54,31 @@ class Xv2Serializer(serializers.Serializer):
     id = serializers.UUIDField()
 
 
+class Person(models.Model):
```
Comment (tfranzel) [inline tests/test_versioning.py:57]: unfortunately you cannot reuse model names in the tests and Person is already taken. use something else

Comment (MissiaL) [reply tests/test_versioning.py]: Done. Sorry for the unprepared code.

**Comment by tfranzel** [comment]
ok i had a look. 

1. your code example never worked since the introduction of versioning in 802814f
2. i believe this is unrelated to #54
3. you try to get a version in the serializer object. since most of the time they are dummy initialized and the request is not available there. you can only get in the view (if at all).

not sure if that is doable at all.

**Comment by tfranzel** [comment]
no actually there is an easy fix

```
    def get_serializer(self):
        return super().get_serializer()
```

in the view. you force to create the serializer with the context and not a dummy serializer. this was thought of as a defensive measure. it will ``get_serializer_class`` and dummy init if ``get_serializer``  is not overridden. that strategy could be changed. worth thinking about that

**Comment by MissiaL** [comment]
It sounds a little impudent for you, but what if you introduce new functionality with the `drop_fields_for_version` or  `spectacular_drop_fields_for_version` parameter ?

This will solve several problems at once.
1) A unique functionality will appear in the library that allows you to delete extra fields depending on the version
2) Allows you to maintain compatibility

I experimented with code
The version can be passed on to
```python
# beware that every access to schema yields a fresh object (descriptor pattern)
operation = view.schema.get_operation(path, path_regex, method, self.registry, version)
```

Further in the code, we may not consider these fields in the parameter depending on the version

@tsouvarev What do you think about that?

**Comment by tfranzel** [comment]
well asking doesn't hurt :smile: but i think this is rather opionionated approach for your case. it might not apply to others. by this we would invent a DRF feature which i think is not the best course of action.

1. i think you got a working version now with my proposed fix
2. compatablility with what exactly?

**Comment by MissiaL** [comment]
> compatablility with what exactly?
I meant that the parameter `spectacular_drop_fields_for_version` will not somehow affect other projects, since it is unique :smile:

**Comment by tfranzel** [comment]
nope sry. since there is a "relatively" easy way to achieve what you want, i would recommend going that way. 

i do not intend to maintain extra functionality that is not generic.

**Comment by tfranzel** [comment]
glad you agree. thanks anyway for the effort.

**Pull request closed without merging**